Sources/Common/DataModel/Line/index.js

```diff
@@ -166,6 +176,7 @@ function vtkLine(publicAPI, model) {
       // one of the parametric coords must be outside 0-1
       if (outObj.t < 0.0) {
         outObj.t = 0.0;
+        outObj.betweenPoints = isBetweenPoints(outObj.t);
```

Comment:
I wouldn't recompute `betweenPoints` here

Reply:
I think it's mandatory because if the value is set to 0.0, then it's considered between points. Or, as outObj.t is inferior to zero right before, the betweenPoint value would probably be false before entering in if.

Reply:
fair enough, I would do the following however:
```
const intersect = intersection(p1, p2, a1, a2, u, v);
outObj.t = u[0];
outObj.betweenPoints = (t >= 0.0 && t <= 1.0);
pcoords[0] = v[0];
...

if (outObj.t < 0.0) {
  outDistance = distanceToLine(p1, a1, a2, x)
  if (outDistance.distance <= tol * tol) {
    outObj.t = 0.0;
    outObj.betweenPoints = true; // intersection is near  p1
    outObj.intersect = 1;
  }
  return outObj;
}
if (outObj.t > 1.0) {
  outDistance = distanceToLine(p2, a1, a2, x)
  if (outDistance.distance <= tol * tol) {
    outObj.t = 1.0;
    outObj.betweenPoints = true; // intersection is near p2
    outObj.intersect = 1;
  }
  return outObj;
}
if (pcoords[1] < 0.) {
  ...
    outObj.betweenPoints = false; // intersection is outside [p1, p2]
  ...
}
if (pcoords[1] < 0.) {
  ...
    outObj.betweenPoints = false; // intersection is outside [p1, p2]
  ...
}

```